app/lib/common/utils/utils.dart

```diff
@@ -184,6 +187,53 @@ String randomString() {
   return base64UrlEncode(values);
 }
 
+Future<void> openAvatar(
+  BuildContext context,
+  WidgetRef ref,
+  String roomId,
+) async {
+  final membership = await ref.watch(roomMembershipProvider(roomId).future);
+  final canUpdateAvatar = membership?.canString('CanUpdateAvatar') == true;
+  final avatarInfo = ref.watch(roomAvatarInfoProvider(roomId));
+
+  if (avatarInfo.avatar != null && context.mounted) {
+    //Open avatar in full screen if avatar data available
+    context.pushNamed(
+      Routes.fullScreenAvatar.name,
+      queryParameters: {'roomId': roomId},
+    );
+  } else if (avatarInfo.avatar == null &&
+      canUpdateAvatar &&
+      context.mounted) {
+    //Change avatar if avatar is null and have relevant permission
+    uploadAvatar(ref, context, roomId);
+  }
+}
+
+Future<void> uploadAvatar(
+  WidgetRef ref,
+  BuildContext context,
+  String roomId,
+) async {
+  final room = await ref.read(maybeRoomProvider(roomId).future);
+  if (room == null || !context.mounted) return;
+  FilePickerResult? result = await FilePicker.platform.pickFiles(
+    type: FileType.image,
+  );
+  if (result == null || result.files.isEmpty) return;
+  try {
+    if (!context.mounted) return;
+    EasyLoading.show(status: L10n.of(context).avatarUploading);
+    final file = result.files.first;
+    if (file.path != null) await room.uploadAvatar(file.path!);
+    // close loading
+    EasyLoading.dismiss();
+  } catch (e, st) {
+    _log.severe('Failed to upload avatar', e, st);
+    EasyLoading.dismiss();
```

Comment:
`EasyLoading.showError` please. Tell the user if we couldn't complete it or they assume it worked and well, it didn't ;) .